### PR TITLE
mgr/dashboard: Zone details showing incorrect data for data pool values and compression info for Storage Classes

### PR DESCRIPTION
**Pull request by aaSharma14**
https://github.com/user-attachments/assets/8fdbc404-03e6-4979-b21f-21bb57a07108

Fixes: https://tracker.ceph.com/issues/67103

Signed-off-by: Aashish Sharma <aasharma@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by afreen23 — CHANGES_REQUESTED** [review]
We should have some tests for these as well - unit test and e2e ones capturing the input fields of the form.

Thanks

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.spec.ts [hunk 1]
```diff
@@ -1,37 +1,109 @@
...
     fixture.detectChanges();
+
+    component.getZonePlacementData('default-placement');
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should set correct values in the form on edit', () => {
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.spec.ts:94]: why dont we add one for create as well :) ?
Meanwhile I can test this too

Comment (aaSharma14) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.spec.ts:92]: Added for create as well

**Review by afreen23 — APPROVED** [review]
This works great now.
The bug which was showing compression for storage class is no more reproducible and I am able to edit/add compression as well as data pools correctly populated.

Thanks @aaSharma14 for adding supporting tests as well.

**Comment by afreen23** [comment]
jenkins test api

**Review by nizamial09 — APPROVED** [review]

**Pull request merged**